pkg/compactor/compactor.go

```diff
@@ -34,8 +34,29 @@ import (
 var (
 	errInvalidBlockRanges = "compactor block range periods should be divisible by the previous one, but %s is not divisible by %s"
 	RingOp                = ring.NewOp([]ring.IngesterState{ring.ACTIVE}, nil)
+
+	DefaultGrouperFactory = func(logger log.Logger, cfg Config, bkt objstore.Bucket, reg prometheus.Registerer, blocksMarkedForDeletion prometheus.Counter, garbageCollectedBlocks prometheus.Counter) compact.Grouper {
```

Comment:
**question(non-blocking):** Is there a circumstance that would require the `cfg` to be a parameter here? Wouldn't the caller be able to initialize and pass the GrouperFactory function with the config pre set? I'm indifferent to leaving it here since it adds flexibility at little cost. Just curious if I was missing something.

Reply:
In the work I've done in the "downstream project" it looked handy having the config passed to the factory 😉